README.md

```diff
@@ -51,9 +53,18 @@ Quick Start
 Requirements:
 - a least docker 26.1
 - Debian bookworm
+- Debian packages `gettext-base` `postgresql` `postgresql-14-postgis-3`
+- [yq](https://github.com/mikefarah/yq/#install)
+
+First, we need to setup the environment and the database. To do so execute the
+following command
+
+```bash
+./script/setup env
+./script/setup db
```

Comment:
I managed to install all dependencies. The `setup.sh` check confirms everything is installed — this is extremely useful btw). I then proceeded with the setup of env and db. The environment setup (`setup.sh env`) went well after resolving a couple things, but when I try to run `setup.sh db`, I encounter the errors below.


`./script/setup.sh db


Job for postgresql@15-main.service failed because the service did not take the steps required by its unit configuration.
See "systemctl status postgresql@15-main.service" and "journalctl -xeu postgresql@15-main.service" for details.
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
	Is the server running locally and accepting connections on that socket?
createdb: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
	Is the server running locally and accepting connections on that socket?
psql: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
	Is the server running locally and accepting connections on that socket?
createdb: error: connection to server on socket "/var/run/postgresql/.s.PGSQL.5432" failed: No such file or directory
	Is the server running locally and accepting connections on that socket?

sudo systemctl status postgresql@15-main.service


× postgresql@15-main.service - PostgreSQL Cluster 15-main
     Loaded: loaded (/lib/systemd/system/postgresql@.service; enabled-runtime; preset: enabled)
     Active: failed (Result: protocol) since Wed 2024-08-14 16:09:38 UTC; 20s ago
   Duration: 1min 40.428s
    Process: 26748 ExecStart=/usr/bin/pg_ctlcluster --skip-systemctl-redirect 15-main start (code=exited, status=1/FAILURE)
        CPU: 102ms

Aug 14 16:09:38 ip-172-31-3-78 postgresql@15-main[26748]: 2024-08-14 16:09:38.403 UTC [26753] CONTEXT:  line 108 of configuration file "/etc/postgresql/15/main/pg_hba.conf"
Aug 14 16:09:38 ip-172-31-3-78 postgresql@15-main[26748]: 2024-08-14 16:09:38.403 UTC [26753] LOG:  specifying both host name and CIDR mask is invalid: ".0.0.0/8"
Aug 14 16:09:38 ip-172-31-3-78 postgresql@15-main[26748]: 2024-08-14 16:09:38.403 UTC [26753] CONTEXT:  line 110 of configuration file "/etc/postgresql/15/main/pg_hba.conf"
Aug 14 16:09:38 ip-172-31-3-78 postgresql@15-main[26748]: 2024-08-14 16:09:38.403 UTC [26753] FATAL:  could not load pg_hba.conf
Aug 14 16:09:38 ip-172-31-3-78 postgresql@15-main[26748]: 2024-08-14 16:09:38.404 UTC [26753] LOG:  database system is shut down
Aug 14 16:09:38 ip-172-31-3-78 postgresql@15-main[26748]: pg_ctl: could not start server
Aug 14 16:09:38 ip-172-31-3-78 postgresql@15-main[26748]: Examine the log output.
Aug 14 16:09:38 ip-172-31-3-78 systemd[1]: postgresql@15-main.service: Can't open PID file /run/postgresql/15-main.pid (yet?) after start: No such file or directory
Aug 14 16:09:38 ip-172-31-3-78 systemd[1]: postgresql@15-main.service: Failed with result 'protocol'.
Aug 14 16:09:38 ip-172-31-3-78 systemd[1]: Failed to start postgresql@15-main.service - PostgreSQL Cluster 15-main.
`

Looking at the postgres service, it seems there’s an issue with the `pg_hba.conf `file. I’m not entirely sure, but it looks like the script might be adding network rules for replication that contain an invalid CIDR mask? When I manually change the CIDR mask for these rules to 0.0.0.0/8, the service starts without any issues.
I might be completely off, but I’ve spent quite some time debugging this and haven’t been able to find the solution.

Reply:
@toch, here's my environment info:

```shell
lsb_release -a
No LSB modules are available.
Distributor ID:	Debian
Description:	Debian GNU/Linux 12 (bookworm)
Release:	12
Codename:	bookworm

docker --version
Docker version 27.1.2, build d01f264

psql (PostgreSQL) 15.8 (Debian 15.8-0+deb12u1)

perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
	LANGUAGE = (unset),
	LC_ALL = (unset),
	LC_CTYPE = "UTF-8",
	LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to a fallback locale ("en_US.UTF-8").
psql (PostgreSQL) 15.8 (Debian 15.8-0+deb12u1)
``` 

Maybe a problem with my locale setting?

Reply:
Thanks @mcrimi 

First of all, I've edited your message to format the pasted output: I've replaced `shell` with `bash`.

do you run it in a VM? on your machine? is it a true debian like distribution, such as ubuntu?

I wasn't able to reproduce the cause of your error.

Could you please:
1. remove the last line in `pg_hba.conf`
2. update the first line of the script `setup.sh` with `#!/bin/bash -xv`
3. run it again, and share the output with me please and also a copy paste of the line added to `pg_hba.conf`

Thanks

Reply:
According to the current output you shared, that it couldn't retrieve correctly the Docker gateway.

could you also share with me the result of the following command please?

```bash
docker network inspect --format='{{(index .IPAM.Config 0).Gateway}}' bridge
```

Reply:
(I didn't have your answer about your environment when submitting my 2 previous comments)

I guess you did run it in a VM?

I think the answer to my last comment will put us on the right track. Also a full `docker network inspect` will be useful. Thanks

Reply:
@toch, this is indeed a VM. Here's the result of the network inspect command:

```bash
docker network inspect --format='{{(index .IPAM.Config 0).Gateway}}' bridge


Cannot connect to the Docker daemon at unix:///Users/mcrimi/.docker/run/docker.sock. Is the docker daemon running?
```

Reply:
Do you run the VM on an Apple?

It seems that the docker service doesn't run. It might be a permission issue. I'll check.

Reply:
@toch, as for the file `pg_hba.conf` edition, it appears that the script added multiple lines due to my multiple retries. Could you guide me towards what to delete?


```
# TYPE  DATABASE        USER            ADDRESS                 METHOD

# "local" is for Unix domain socket connections only
local   all             all                                     peer
# IPv4 local connections:
host    all             all             127.0.0.1/32            scram-sha-256
# IPv6 local connections:
host    all             all             ::1/128                 scram-sha-256
# Allow replication connections from localhost, by a user with the
# replication privilege.
local   replication     all                                     peer
host    replication     all             127.0.0.1/32            scram-sha-256
host    replication     all             ::1/128                 scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
```

Reply:
@toch, no Apple. I'm running this VM in an AWS EC2 instance.

Reply:
@mcrimi you can remove the last 9 lines

```
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
host    all     all     .0.0.0/8        scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
```

could you share the spec of the VM? thanks

It's very weird that you have such path `unix:///Users/mcrimi/.docker/run/docker.sock.` It's like you're on a Apple machine.

Do you run the command in the VM?

Reply:
@toch , so sorry!  Got disconnected when I executed the `docker network inspect --format='{{(index .IPAM.Config 0).Gateway}}' bridge` and that's why it got executed in my machine 🤦🏻‍♂️

Here's the actual result:

```bash
docker network inspect --format='{{(index .IPAM.Config 0).Gateway}}' bridge
172.17.0.1
```

Reply:
ok this looks right.

notice the IP address `172.17.0.1` that's the one we're looking for.

You get `.0.0.0/8` when the command `docker network inspect --format='{{(index .IPAM.Config 0).Gateway}}' bridge` doesn't return you anything.

Considering that we can retrieve the Docker gateway IP address, it's weird that we don't get it right.

In your `pg_hba.conf`, you have sometimes the right subnet `172.0.0.0/8`. Have you added it yourself or has it been added by the script?

Reply:
@toch - `#!/bin/bash -xv` did the trick it seems.

This is my `pg_hba.conf` now.
```bash
# TYPE  DATABASE        USER            ADDRESS                 METHOD

# "local" is for Unix domain socket connections only
local   all             all                                     peer
# IPv4 local connections:
host    all             all             127.0.0.1/32            scram-sha-256
# IPv6 local connections:
host    all             all             ::1/128                 scram-sha-256
# Allow replication connections from localhost, by a user with the
# replication privilege.
local   replication     all                                     peer
host    replication     all             127.0.0.1/32            scram-sha-256
host    replication     all             ::1/128                 scram-sha-256
host    all     all     172.0.0.0/8     scram-sha-256
``` 

I checked and DB's and roles are created, thanks so much!

Reply:
Noted. Will do. In that case should I do `./script/openhexa.sh purge` and start over?

Reply:
I can also do a fresh install...

Reply:
both works. but I'd go for a fresh install as it might be linked to a change in a dep (very likely docker).

Also if you give me access, I can have a look myself.

Reply:
@toch Moved forward with the fresh install. using your latest commit. I think I've installed everything but the `setup.sh check` points some errors.

```bash
./script/setup.sh check
Check installation:
- file .env ... not present or not readable
- file .env.dist ... present
- file compose.yml ... present
- file debian/openhexa.service ... present
- file script/common_functions.sh ... present
- file script/openhexa.sh ... present
- file script/setup.sh ... present
- Systemd service openhexa.service ... not required (run locally)
- package gettext-base ... installed
- package openssl ... installed
- package postgresql ... installed
- package postgresql-*-postgis-3 ... installed
- binary yq ... installed
- docker 26+ ... installed
- docker engine is ... not running.
- postgresql service is ... running
``` 

Not quite sure what to do with the missing `.env` file.  As for the docker engine, I looks like its running:

```basht
sudo docker info
Client: Docker Engine - Community
 Version:    27.1.2
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.16.2
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.29.1
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 1
  Running: 0
  Paused: 0
  Stopped: 1
 Images: 1
 Server Version: 27.1.2
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 8fc6bcff51318944179630522a095cc9dbf9f353
 runc version: v1.1.13-0-g58aa920
 init version: de40ad0
 Security Options:
  apparmor
  seccomp
   Profile: builtin
  cgroupns
 Kernel Version: 6.1.0-23-cloud-amd64
 Operating System: Debian GNU/Linux 12 (bookworm)
 OSType: linux
 Architecture: x86_64
 CPUs: 4
 Total Memory: 15.45GiB
 Name: ip-172-31-14-118
 ID: 8dbc62bf-db31-448a-8ec6-0f2ef19dadcd
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

``` 

I'll share access to this machine via DM.